# KAPTEN Campaign Analysis - Promocode

The objective of this project is to understand whether a ride promocode marketing campaign by Kapten was "good" & worthwhile. The promocode was emailed only to current users of Kapten. It multiplied loyalty points accumulated during this period by 5 times. Users are grouped by: Red  (0) Silver (1) Gold (2) Platinum (3) loyalty statuses, where each number is points per euro spent.  

The campaign ocurred from the 17th to the 21st of May. Data used is from a random subset of its ride data which has been altered for privacy purposes.   

If it was a good campaign, the company will have gained a ROI return on investment of 25 - 50% or above: a range which current [marketing industry standards](https://themarketingsquare.com/2012/04/whats-a-good-advertising-roi/) would consider large enough to be worthwhile. Also, the conversion rate for promocodes claimed vs. rides won would coincide with industry averages. In addition, the company will have increased customer loyalty. (See Analysis at end)

In [1]:
import pandas as pd
import numpy as py

### Rides Dataset

In [2]:
df1 = pd.read_csv('rides.csv', delimiter=',')
df1

ride_id                           user_id  \
0        357a545ef665da69dd86ac33ae817f3a  c6f3a631ae33a84c343dd9c553889337   
1        5714536c4b4b1b10c4cd8ac907537044  0bad5fc7936b072d8de8749b9dd608c6   
2        5c8f5dec570b6597090f8b30ea33b0af  0bad5fc7936b072d8de8749b9dd608c6   
3        0f37be24f23a15746eb8774e7a1bd934  37169a3d6636e14afabb1dbe01f5c83e   
4        67f054abb8b60132dd7c19cb30e74e10  b89d8219faa82966c7b33ee69b484ad2   
...                                   ...                               ...   
1445659  790a5ce8fb2253953f99f08978d792ba  53e3cb5a4410c965300760b69f61a1c4   
1445660  014a51a17ea56cb91040c7273fd50b15  4c3df7a611fe860a54fb0799e0001903   
1445661  510083d4ed088808455d6db48f60e3ca  7ca9825c2744adf0ec4505e379a28b73   
1445662  df315daa45794ac80001878f50e6f363  8f100f37bb092089f418936edc0f94c9   
1445663  44b1e577a76dc15a4e5134275c3eb8ca  b2501c4161c76f90466c7ec3a7e29b3e   

         loyalty.status                        booking_id  isGuest  \
0                     1  1ea36f95a86a5a876456ce66587f4882    False   
1                     1                               NaN    False   
2                     1                               NaN    False   
3                     0                               NaN    False   
4                     3                               NaN    False   
...                 ...                               ...      ...   
1445659               2                               NaN    False   
1445660               1                               NaN    False   
1445661               1                               NaN    False   
1445662               3                               NaN    False   
1445663               3                               NaN    False   

         loyalty_points_earned price.loyalty.points_spent  price_nominal  \
0                            0                        NaN           5.48   
1                            0                        NaN           4.72   
2                            5                        NaN           4.72   
3                            8                        NaN           7.68   
4                            7                        NaN           4.00   
...                        ...                        ...            ...   
1445659                      8                        NaN           5.95   
1445660                      9                        NaN           8.85   
1445661                      6                        NaN           4.10   
1445662                      0                        NaN           7.72   
1445663                      8                        NaN           4.67   

         price_after_discount          state           quote_date  
0                        5.80  not_completed  2017-03-01 11:07:49  
1                        4.72  not_completed  2017-03-02 22:45:02  
2                        4.72      completed  2017-03-02 22:45:22  
3                        7.68      completed  2017-03-03 00:22:45  
4                        4.00      completed  2017-03-04 03:32:15  
...                       ...            ...                  ...  
1445659                  5.95      completed  2017-05-19 22:59:30  
1445660                  8.85      completed  2017-05-20 05:14:17  
1445661                  4.10      completed  2017-05-21 20:57:58  
1445662                  7.72  not_completed  2017-05-22 12:59:29  
1445663                  4.67      completed  2017-05-22 16:28:07  

[1445664 rows x 11 columns]

### Events Dataset

In [3]:
df2 = pd.read_csv('events.csv', delimiter=',')  
df2

event_id                           ride_id  \
0       457113065c424d25eff17823ad888162                               NaN   
1       06e5ceb932743b583dc4349f617d1fde                               NaN   
2       f08b5f110a9e2e9c3eac3f03e8ad801b                               NaN   
3       4265bc09df57f4fb6991f832b24e643c                               NaN   
4       4a5d1f58698acb12ca4da6b24535693d                               NaN   
...                                  ...                               ...   
292304  a950539ed8b494d53123eac51faa542d  95a78c37f93390c7c8a09c10843ae44b   
292305  888680b75efde5d0af32c33e5dfa2531  c83efd478f8991b73db5bbaa0523e265   
292306  3da09eb0dd1b86fea4d79557c51abaf3  aaf154183a33b3ecb513b488e56ae217   
292307  c5c9ae780440419b40efd0b1db7d811f  3442ac80e226a3834e00aabe9faf8b9f   
292308  daf296a54364257aa702dbfa92b10a8a  d0e6d68c457d4211c0d707211d52e350   

                       type                           user_id  \
0       other_coupon_action  c045abae8a3110ea6cec54b2f49f9f2d   
1            coupon_created  c33cc932b7a4feb6f51d7ed1077faa05   
2            coupon_created  da03b00f7dd32fa31b800ecb762ef1df   
3            coupon_created  0928a1d559774b05be5e6926920d2873   
4            coupon_created  960025162805d2c5942334c5ed200528   
...                     ...                               ...   
292304  other_coupon_action  ecc32d29b014a0e40c2f569923c9f915   
292305  other_coupon_action  a79c1ef0d2e0498dde420979151b2e6b   
292306  other_coupon_action  b343d8fede8563bf164777d9a3cb644a   
292307  other_coupon_action  655541972e25436abbfbe97db0ea0bcc   
292308  other_coupon_action  d3bee7df44b558e2392602cf5a6202d9   

                          promocode_name  
0       89f13478125cee693ce7f76fedf01808  
1       8041613eff4408b9268b66430cf5d9a1  
2       308dd493382b32ebf486da92fdeaf38d  
3       c4afe8e1914a1f612ca82d59ec72c771  
4       308dd493382b32ebf486da92fdeaf38d  
...                                  ...  
292304  a74bc31c9f97fad86c213835094be8a8  
292305  5e48c468d9f81fb0d5258e8cb39d3dca  
292306  6ae862a7af2164c5333b6189e351658d  
292307  a74bc31c9f97fad86c213835094be8a8  
292308  fe62782812fd85a64a14e69c5f6f0c7b  

[292309 rows x 5 columns]

### Both Datasets

In [4]:
df = df1.merge(df2, on='ride_id', how='outer').rename(columns={'user_id_x':'user_id'}).drop(['user_id_y'], axis=1)
df

# Merges. Drops extra user_id column as result of merge

ride_id                           user_id  \
0        357a545ef665da69dd86ac33ae817f3a  c6f3a631ae33a84c343dd9c553889337   
1        5714536c4b4b1b10c4cd8ac907537044  0bad5fc7936b072d8de8749b9dd608c6   
2        5c8f5dec570b6597090f8b30ea33b0af  0bad5fc7936b072d8de8749b9dd608c6   
3        0f37be24f23a15746eb8774e7a1bd934  37169a3d6636e14afabb1dbe01f5c83e   
4        67f054abb8b60132dd7c19cb30e74e10  b89d8219faa82966c7b33ee69b484ad2   
...                                   ...                               ...   
1698361  95a78c37f93390c7c8a09c10843ae44b                               NaN   
1698362  c83efd478f8991b73db5bbaa0523e265                               NaN   
1698363  aaf154183a33b3ecb513b488e56ae217                               NaN   
1698364  3442ac80e226a3834e00aabe9faf8b9f                               NaN   
1698365  d0e6d68c457d4211c0d707211d52e350                               NaN   

         loyalty.status                        booking_id isGuest  \
0                   1.0  1ea36f95a86a5a876456ce66587f4882   False   
1                   1.0                               NaN   False   
2                   1.0                               NaN   False   
3                   0.0                               NaN   False   
4                   3.0                               NaN   False   
...                 ...                               ...     ...   
1698361             NaN                               NaN     NaN   
1698362             NaN                               NaN     NaN   
1698363             NaN                               NaN     NaN   
1698364             NaN                               NaN     NaN   
1698365             NaN                               NaN     NaN   

         loyalty_points_earned price.loyalty.points_spent  price_nominal  \
0                          0.0                        NaN           5.48   
1                          0.0                        NaN           4.72   
2                          5.0                        NaN           4.72   
3                          8.0                        NaN           7.68   
4                          7.0                        NaN           4.00   
...                        ...                        ...            ...   
1698361                    NaN                        NaN            NaN   
1698362                    NaN                        NaN            NaN   
1698363                    NaN                        NaN            NaN   
1698364                    NaN                        NaN            NaN   
1698365                    NaN                        NaN            NaN   

         price_after_discount          state           quote_date  \
0                        5.80  not_completed  2017-03-01 11:07:49   
1                        4.72  not_completed  2017-03-02 22:45:02   
2                        4.72      completed  2017-03-02 22:45:22   
3                        7.68      completed  2017-03-03 00:22:45   
4                        4.00      completed  2017-03-04 03:32:15   
...                       ...            ...                  ...   
1698361                   NaN            NaN                  NaN   
1698362                   NaN            NaN                  NaN   
1698363                   NaN            NaN                  NaN   
1698364                   NaN            NaN                  NaN   
1698365                   NaN            NaN                  NaN   

                                 event_id                 type  \
0                                     NaN                  NaN   
1                                     NaN                  NaN   
2                                     NaN                  NaN   
3                                     NaN                  NaN   
4                                     NaN                  NaN   
...                                   ...                  ...   
1698361  a950539ed8b494d53123eac51faa542d  other_cou

### USERS who Claimed Promocode

(Given claimed includes 'other_coupon_action' and 'coupon_created')

In [5]:
Promocodes_Claimed = df[(df.promocode_name == 'promocode') & (df.type != '-')]
Promocodes_Claimed

#The (df.type != '-') is made to include BOTH 'other_coupon_action' AND 'coupon_created' per stated assumption

ride_id                           user_id  \
252      4151effb5a7998e13a60a4710c238d80  73dd38be4f6a294a9caad81643a405b2   
534      c3338e5d980721efc20e6c6f77374d4b  a2750039f85afe319ad0f81ca96d258b   
595      d495cdcb6b0615f0fd2620fc29b5f3fc  367da9d6fcee331541a3f8771d07dc97   
720      939d9b22825fb7cef15d852ba93b7b8d  99d6ae8df36639e25efa88f3c141e65b   
775      03b2151a85c0069c475e27365ddf2fc5  7d4e5564331966503bad3a5736a47c58   
...                                   ...                               ...   
1693760  c92f892dc3fef4b0df26040037aae321                               NaN   
1693761  8fa750a2a1f5a8ea1987032988643d6a                               NaN   
1694095  fed18d07f589845e887ebc31e9c17d30                               NaN   
1694096  04cb9172855b750cebf14b96f9679358                               NaN   
1694097  85cb9e751adf8568032af958816b4787                               NaN   

         loyalty.status booking_id isGuest  loyalty_points_earned  \
252                 3.0        NaN   False                    9.0   
534                 1.0        NaN   False                    8.0   
595                 2.0        NaN   False                    9.0   
720                 2.0        NaN   False                   11.0   
775                 2.0        NaN   False                    7.0   
...                 ...        ...     ...                    ...   
1693760             NaN        NaN     NaN                    NaN   
1693761             NaN        NaN     NaN                    NaN   
1694095             NaN        NaN     NaN                    NaN   
1694096             NaN        NaN     NaN                    NaN   
1694097             NaN        NaN     NaN                    NaN   

        price.loyalty.points_spent  price_nominal  price_after_discount  \
252                            NaN           4.95                  4.95   
534                            NaN           7.87                  7.87   
595                            NaN           5.14                  4.99   
720                            NaN           6.56                  5.74   
775                            NaN           4.00                  4.00   
...                            ...            ...                   ...   
1693760                        NaN            NaN                   NaN   
1693761                        NaN            NaN                   NaN   
1694095                        NaN            NaN                   NaN   
1694096                        NaN            NaN                   NaN   
1694097                        NaN            NaN                   NaN   

             state           quote_date                          event_id  \
252      completed  2017-05-19 22:34:29  ab92a0c0140e050331e10ee190df934e   
534      completed  2017-05-17 21:21:05  521bab972d6757ffd51a78eef914df8c   
595      completed  2017-05-17 23:06:12  69a62320e9ca6f26428ae8303e649aab   
720      completed  2017-05-18 12:32:08  83e8951099d01e2feac0e3690d204a63   
775      completed  2017-05-22 12:07:18  00df65a63fb73333b6e756d760fe4aea   
...            ...                  ...                               ...   
1693760        NaN                  NaN  8f9420c53e5be5e40da8a1ecac498853   
1693761        NaN                  NaN  13b8fd3787fe7029d893bfdaa90312d6   
1694095        NaN                  NaN  b759a6cec12ffbdd4b181fe816cba50e   
1694096        NaN                  NaN  a39669f71d6e4ddaeb348dbadf0fa3e4   
1694097        NaN                  NaN  8c2aaa76961716a867a905ccab99e484   

                        type promocode_name  
252      other_coupon_action      promocode  
534      other_coupon_action      promocode  
595      other_coupon_action      promocode  
720      other_coupon_action      promocode  
775      other_coupon_action      promocode  
...                      ...            ...  
1693760  other_coupon_action      promocode  
1693761  other_coupon_action      promocode  
1694095  other_c

In [7]:
Promocodes_Claimed_Count = len(Promocodes_Claimed)
Promocodes_Claimed_Count

# Including Repeats User_Id entries since some users had multiple Rides. 
# len insetad of count because some user_id values are missing/null 

41988

In [37]:
FindNull_UserIDs = Promocodes_Claimed.user_id.isnull().value_counts()
FindNull_UserIDs

True     25117
False    16871
Name: user_id, dtype: int64

In [9]:
Nulls = 25117

In [10]:
UniqueUsers_Promocodes_Claimed = Promocodes_Claimed.user_id.value_counts()
UniqueUsers_Promocodes_Claimed

# Count breakdown of how many times each UNIQUE user_id appears.

fbab7fb59b5f75260a011a9fe0c24b2b    24
69455dc18fa192f2179ff9388a70ad3f    22
3033d34c0416c8a139fd1b8311e48564    21
442c00ea917fa5089fdc81759e13ac14    21
5f5fe5ec9a4740debd45b71478a7e393    21
                                    ..
1ef11ce5187ca822f0242d7fa2884dff     1
0d20838e22c65e27678148a4f8457a02     1
d1527423cbb668bd2897ae97dd3555f3     1
0315b58fae75f74e091567ea1d8f733a     1
0677e0e376f4f4eff549adc9422a78c3     1
Name: user_id, Length: 6758, dtype: int64

In [11]:
UUPC_Total = len(UniqueUsers_Promocodes_Claimed)  
UUPC_Total

# From count breakdown for each user_ID above, outputs total unique BEFORE missing/NULL user_ID count added

6758

In [41]:
UUPC_Total_N = UUPC_Total + Nulls
UUPC_Total_N

# Adds nulls assumed to be UNIQUE user_ids for the sake of later obtaining an estimated unique conversion rate

31875

##### RESULT -- 31,875 Unique Users out of 41,988 Users including Repeats claimed the promocode. 

### USER Winners

(Assumes Loyalty_points_earned column not needed per instructions stating: "the bonus points were credited with an adhoc script at the end of the campaign & additionnal points cannot be seen in the data set")

(Given other_coupon_action & coupon_created implies coupon claimed) 

(If'promocode' used ONLY for May 17 - May 20 campaign, then do not need to use a timestamp in eligibility criteria. Instructions stated: "The promocode to study has the name "promocode")

In [13]:
Eligible = df[(df.promocode_name == 'promocode') & (df.state == 'completed') & (df.type != '-') & (df.isGuest != 'True')]
Eligible

#The (df.type != '-') is used to include other_coupon_action' AND 'coupon created' 

ride_id                           user_id  \
252      4151effb5a7998e13a60a4710c238d80  73dd38be4f6a294a9caad81643a405b2   
534      c3338e5d980721efc20e6c6f77374d4b  a2750039f85afe319ad0f81ca96d258b   
595      d495cdcb6b0615f0fd2620fc29b5f3fc  367da9d6fcee331541a3f8771d07dc97   
720      939d9b22825fb7cef15d852ba93b7b8d  99d6ae8df36639e25efa88f3c141e65b   
775      03b2151a85c0069c475e27365ddf2fc5  7d4e5564331966503bad3a5736a47c58   
...                                   ...                               ...   
1445411  577b4ce4b7fce422499115ec204a23e3  689e4b1d95bae82060e13b6db45886b8   
1445490  de55b52911c5522342be35260a39dab7  7ca9825c2744adf0ec4505e379a28b73   
1445764  c6b2e7c0abf90249a56feafa8d70d0c8  fe19379e63d5b98eaf33facc024a211d   
1445847  510083d4ed088808455d6db48f60e3ca  7ca9825c2744adf0ec4505e379a28b73   
1445849  44b1e577a76dc15a4e5134275c3eb8ca  b2501c4161c76f90466c7ec3a7e29b3e   

         loyalty.status booking_id isGuest  loyalty_points_earned  \
252                 3.0        NaN   False                    9.0   
534                 1.0        NaN   False                    8.0   
595                 2.0        NaN   False                    9.0   
720                 2.0        NaN   False                   11.0   
775                 2.0        NaN   False                    7.0   
...                 ...        ...     ...                    ...   
1445411             3.0        NaN   False                   14.0   
1445490             1.0        NaN   False                    5.0   
1445764             1.0        NaN   False                    6.0   
1445847             1.0        NaN   False                    6.0   
1445849             3.0        NaN   False                    8.0   

        price.loyalty.points_spent  price_nominal  price_after_discount  \
252                            NaN           4.95                  4.95   
534                            NaN           7.87                  7.87   
595                            NaN           5.14                  4.99   
720                            NaN           6.56                  5.74   
775                            NaN           4.00                  4.00   
...                            ...            ...                   ...   
1445411                        NaN           8.23                  8.23   
1445490                        NaN           4.90                  4.90   
1445764                        NaN           4.40                  4.40   
1445847                        NaN           4.10                  4.10   
1445849                        NaN           4.67                  4.67   

             state           quote_date                          event_id  \
252      completed  2017-05-19 22:34:29  ab92a0c0140e050331e10ee190df934e   
534      completed  2017-05-17 21:21:05  521bab972d6757ffd51a78eef914df8c   
595      completed  2017-05-17 23:06:12  69a62320e9ca6f26428ae8303e649aab   
720      completed  2017-05-18 12:32:08  83e8951099d01e2feac0e3690d204a63   
775      completed  2017-05-22 12:07:18  00df65a63fb73333b6e756d760fe4aea   
...            ...                  ...                               ...   
1445411  completed  2017-05-21 01:32:35  984650000a2ff4b9d7051177a7868607   
1445490  completed  2017-05-19 00:38:23  d272c2932fda2dd46f7f5711d9878b43   
1445764  completed  2017-05-21 22:17:07  a75449aff97de25d9a34af7056dfa304   
1445847  completed  2017-05-21 20:57:58  b672a487c8c49b3f734e20893fa63261   
1445849  completed  2017-05-22 16:28:07  ac2f2617a9ce62beed293e3a1cbbbeb4   

                        type promocode_name  
252      other_coupon_action      promocode  
534      other_coupon_action      promocode  
595      other_coupon_action      promocode  
720      other_coupon_action      promocode  
775      other_coupon_action      promocode  
...                      ...            ...  
1445411  other_coupon_action      promocode  
1445490  other_coupon_action      promocode  
1445764  other_c

In [14]:
Winner_Count = Eligible.user_id.count()
Winner_Count

16713

In [15]:
UniqueUsers_Winners = Eligible.user_id.value_counts()
UniqueUsers_Winners

fbab7fb59b5f75260a011a9fe0c24b2b    24
69455dc18fa192f2179ff9388a70ad3f    21
3033d34c0416c8a139fd1b8311e48564    21
f9713567071fc2620c3896649a31ed5a    21
442c00ea917fa5089fdc81759e13ac14    20
                                    ..
63f5bf48e469c837e06a09606a3a0b61     1
f1f09675697de02550194a2d6a14aecb     1
0743de05cdd7dcc7a62a3e2c4b931e10     1
cbd129ba01569caef68f6c843f51a062     1
7ee716fcdcfb9ad56fcb3f376cc695c5     1
Name: user_id, Length: 6737, dtype: int64

In [16]:
UUW_Total = len(UniqueUsers_Winners)
UUW_Total

6737

##### RESULT --  Winners: 6,737 UNIQUE Users out of 16,713 Users rides including repeats by same user

### Conversion rate:  Percent of Winners over who only Claimed Promocode

(Given claimed includes 'other_coupon_action')

In [33]:
'{:,.2%}'.format(round(Winner_Count / Promocodes_Claimed_Count,2)) + ' of users- including repeat rides by same user- who claimed the code won points'

'40.00% of users- including repeat rides by same user- who claimed the code won points'

In [18]:
'{:,.2%}'.format(round(UUW_Total / UUPC_Total_N,2)) + ' Percent of UNIQUE users who claimed the code won points.'

'21.00% Percent of UNIQUE users who claimed the code won points.'

### Number of 'Promocode' Rides

(Assumes state of ride does not need to be 'completed')

In [19]:
Events_Rides = df.filter(['promocode_name'])

# Step 1, filter by promocode_name

In [20]:
Promocode_Rides = Events_Rides[(Events_Rides.promocode_name == 'promocode')].count()
Promocode_Rides

# Step 2, count only promocode

promocode_name    41988
dtype: int64

##### RESULT --  41,988 rides taken with the 'promocode' event campaign

### User rides by Loyalty Status:  Winners vs. Non-winners

(Assumes repeat rides from the same user are also counted)

In [36]:
Total_Loyalty_Statuses = df.groupby('loyalty.status').count().filter(['user_id']).rename(columns={'user_id':'user_id_count'})
Total_Loyalty_Statuses

user_id_count
loyalty.status               
0.0                    404732
1.0                    380783
2.0                    353215
3.0                    307120

##### Winner User rides by Loyalty Status

In [22]:
Winners_Loyalty = Eligible.groupby('loyalty.status').count().filter(['user_id']).rename(columns={'user_id':'user_id_count'})
Winners_Loyalty

user_id_count
loyalty.status               
0.0                      1728
1.0                      4506
2.0                      5675
3.0                      4804

##### Non-Winner User Rides by Loyalty Status

In [24]:
Non_Winners = Total_Loyalty_Statuses - Winners_Loyalty
Non_Winners

user_id_count
loyalty.status               
0.0                    403004
1.0                    376277
2.0                    347540
3.0                    302316

#### Additional Summary Statistics

In [28]:
df.describe().round(1).drop('count')

# describes both datasets merged

loyalty.status  loyalty_points_earned  price_nominal  \
mean             1.4                    6.2            6.1   
std              1.1                    4.3            1.9   
min              0.0                    0.0            0.0   
25%              0.0                    4.0            4.6   
50%              1.0                    7.0            5.6   
75%              2.0                    9.0            7.1   
max              3.0                   59.0          220.8   

      price_after_discount  
mean                   5.8  
std                    2.2  
min                    0.0  
25%                    4.4  
50%                    5.5  
75%                    7.0  
max                  220.8

In [29]:
df1.describe().round(1).drop('count')

# describes RIDE dataset

loyalty.status  loyalty_points_earned  price_nominal  \
mean             1.4                    6.2            6.1   
std              1.1                    4.3            1.9   
min              0.0                    0.0            0.0   
25%              0.0                    4.0            4.6   
50%              1.0                    7.0            5.6   
75%              2.0                    9.0            7.1   
max              3.0                   59.0          220.8   

      price_after_discount  
mean                   5.8  
std                    2.2  
min                    0.0  
25%                    4.4  
50%                    5.5  
75%                    7.0  
max                  220.8

In [30]:
df2.describe().round(1)

# describes EVENT dataset

event_id                           ride_id  \
count                             292309                            102431   
unique                            292309                            101754   
top     01a2d5fc880c0fe9b968b3a0cf063478  e8619b9cade5e0867788f9f4bfdb7ed3   
freq                                   1                                 4   

                  type                           user_id  \
count           292309                            292309   
unique               2                             74335   
top     coupon_created  6879c8222c55a5f13e0841999ce604ad   
freq            172175                               137   

                          promocode_name  
count                             292309  
unique                              4350  
top     9a03c798c04a6839996c26c8d34fa835  
freq                               66938

### Was it a Good campaign? 

**Quantitiative: Toward understanding ROI, Conversion rate, etc.**

What multiplying loyalty points x5 cost the company for this campaign (1 pt gained per ride) 

0.1€ per point x 5 = 0.5€ per Winning Ride 

0.5€ x Winning Ride Count = Loyalty Campaign Cost

0.5€ x 16,713 = 8,356.5€ 

(Campaign Attributable Profit - Loyalty Campaign Cost) / Loyalty Campaign Cost * 100 = ROI

(Campaign Attributable Profit  - 8,356.5€ ) / 8,356.5€ * 100 = ROI

It would be useful to compare what changed with trends before the campaign began. The goal would be to calculate the additional profit gained as the result of the promocode campaign. In order to fully assess & calculate return on investment/ROI one would require further data.  

##### The Conversion Rate:

**40%** of user rides which claimed the code won points-- **21%** were UNIQUE users. Good conversion for promocodes & coupons vary greatly depending on industry & type of promotion, e.g. 7% - 50%+. Deeper research is necessary.

The other **60 - 79%?** Several possibilities. Examples: A user may have tried to use a claimed code outside the promotion timeframe. Or, a user may have reserved a ride they did not take. There could also have been a misunderstanding. However, that these users noticed a promocode at all could drive future loyalty.


    

**Qualitative:**

Customer Loyalty is important to whether a campaign is good. Loyalty can ultimately increase the quantity of rides from existing users & add new users. In addition, it increases brand legacy & popularity.  

In the loyalty count breakdown, those already loyal (had pts before campaign) were most responsive overall. This is predictable buyer behavior. However, those with status 2 outnumbered 3. Perhaps members who already reached status 3 (or could afford to) would be less incentivized by discounts. Data after the campaign is required to further assess loyalty impact.



### Conclusion: 

Whether this campaign was good depends on the profit attributed to the campaign, the resulting ROI & the client loyalty impact (data outside the scope of this study.) One can benchmark the ROI against the range of what can be considered good by current marketing standards. For loyalty, one could analyze recent past trends & future growth following the campaign in addition to comparing it with industry standards.
